rest/aggregates/models.go

```diff
@@ -0,0 +1,216 @@
+	Locale            string  `json:"g,omitempty"`
+	OfficialOpenPrice float64 `json:"op,omitempty"`
+	AverageSize       float64 `json:"z,omitempty"`
+	AccumulatedVolume float64 `json:"av,omitempty"`
+	StartTimestamp    int64   `json:"s,omitempty"`
+	EndTimestamp      int64   `json:"e,omitempty"`
+}
+
+type AggsResponse struct {
+	client.BaseResponse
+	Ticker       string      `json:"ticker,omitempty"`
+	QueryCount   int         `json:"queryCount"`
+	ResultsCount int         `json:"resultsCount"`
+	Adjusted     bool        `json:"adjusted"`
+	Aggs         []Aggregate `json:"results,omitempty"`
+}
+
+func (r *AggsResponse) UnmarshalJSON(data []byte) error {
+	type aggsResponse AggsResponse
+	var v aggsResponse
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	*r = AggsResponse(v)
+	return nil
+}
+
+type GetParams struct {
+	Ticker      string
+	Multiplier  int
+	Resolution  string
+	From        time.Time
+	To          time.Time
+	QueryParams *GetQueryParams
+}
+
+type GetQueryParams struct {
+	Sort     string
+	Limit    int32
+	Adjusted bool
+	Explain  bool
+}
+
+func (p GetParams) Path() map[string]string {
+	return map[string]string{
+		"ticker":     p.Ticker,
+		"multiplier": fmt.Sprint(p.Multiplier),
+		"resolution": fmt.Sprint(p.Resolution),
+		"from":       fmt.Sprint(p.From.UnixMilli()),
+		"to":         fmt.Sprint(p.To.UnixMilli()),
+	}
+}
+
+func (p GetParams) Query() map[string]string {
+	q := map[string]string{}
+	if p.QueryParams == nil {
+		return q
+	}
+
+	if p.QueryParams.Sort != "" {
+		q["sort"] = p.QueryParams.Sort
+	}
+
+	if p.QueryParams.Limit != 0 {
+		q["limit"] = strconv.FormatInt(int64(p.QueryParams.Limit), 10)
+	}
+
+	if !p.QueryParams.Adjusted {
+		q["adjusted"] = "false"
+	}
+
+	if p.QueryParams.Explain {
+		q["explain"] = "true"
+	}
+
+	return q
+}
+
+type GetPreviousCloseParams struct {
+	Ticker      string
+	QueryParams *GetPreviousCloseQueryParams
+}
+
+type GetPreviousCloseQueryParams struct {
+	Adjusted bool
+}
+
+func (p GetPreviousCloseParams) Path() map[string]string {
+	return map[string]string{
+		"ticker": p.Ticker,
+	}
+}
+
+func (p GetPreviousCloseParams) Query() map[string]string {
+	q := map[string]string{}
+	if p.QueryParams == nil {
+		return q
+	}
+
+	if !p.QueryParams.Adjusted {
+		q["adjusted"] = "false"
+	}
+
+	return q
+}
+
+type GetGroupedDailyParams struct {
+	Locale      string
+	MarketType  string
+	Date        time.Time
+	QueryParams *GetGroupedDailyQueryParams
+}
+
+type GetGroupedDailyQueryParams struct {
+	Adjusted bool
+}
+
+func (p GetGroupedDailyParams) Path() map[string]string {
+	return map[string]string{
+		"locale":     p.Locale,
+		"marketType": p.MarketType,
+		"date":       p.Date.Format("2006-01-02"),
+	}
+}
+
+func (p GetGroupedDailyParams) Query() map[string]string {
+	q := map[string]string{}
+	if p.QueryParams == nil {
+		return q
+	}
+
+	if !p.QueryParams.Adjusted {
+		q["adjusted"] = "false"
+	}
+
+	return q
+}
+
+type DailyOpenCloseResponse struct {
```

Comment:
The endpoint for `Daily Open/Close` provides a different response than the other Aggregates endpoints.

Also, the crypto-specific Daily Open/Close endpoint has a different set of path params and a different response.
https://polygon.io/docs/crypto/get_v1_open-close_crypto__from___to___date

Just something to keep in mind.